src/kagglehub/datasets.py

```diff
@@ -60,3 +72,65 @@ def dataset_upload(
     )
 
     create_dataset_or_version(h, tokens, version_notes)
+
+
+def load_dataset(
+    adapter: KaggleDatasetAdapter,
+    # In the form of {owner_slug}/{dataset_slug} or {owner_slug}/{dataset_slug}/versions/{version_number}
+    handle: str,
+    path: str,
+    *,
+    # Kaggle doesn't currently support test/train splits natively like Hugging Face does, so users
+    # would specify here if they select the 'hugging_face' adapter.
```

Comment:
We should use **kwargs and just pass that to the underlying method. This way,  can support everything the underlying method supports (e.g. pd.read_csv).

Reply:
I was a bit worried about how this scales, so I'll give this a try. I think my thought was that it'd be nice to validate upfront before triggering a user download, but maybe that pushes too much maintenance cost onto us.